src/battle_ai_util.c

```diff
@@ -2954,19 +2954,42 @@ bool32 IsWakeupTurn(u32 battler)
 bool32 AnyPartyMemberStatused(u32 battlerId, bool32 checkSoundproof)
 {
     struct Pokemon *party;
-    u32 i;
+    u32 i, battlerOnField1, battlerOnField2;
 
     if (GetBattlerSide(battlerId) == B_SIDE_PLAYER)
         party = gPlayerParty;
     else
         party = gEnemyParty;
 
-    for (i = 0; i < PARTY_SIZE; i++)
+    if (gBattleTypeFlags & BATTLE_TYPE_DOUBLE)
     {
-        if (checkSoundproof && GetMonAbility(&party[i]) == ABILITY_SOUNDPROOF)
-            continue;
+        battlerOnField1 = gBattlerPartyIndexes[battlerId];
+        battlerOnField2 = gBattlerPartyIndexes[GetBattlerAtPosition(BATTLE_PARTNER(GetBattlerPosition(battlerId)))];
+        // Check partner's status
+        if ((GetMonData(&party[battlerOnField2], MON_DATA_STATUS) != STATUS1_NONE)
+         && (GetBattlerAbility(battlerOnField2) != ABILITY_SOUNDPROOF
```

Comment:
I would make 2 bools for those things since they are checked twice and would make the code a little bit cleaner

Reply:
Sadly these aren't checked twice, it's a different battler in each case.